src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValueTaskExtensions.cs

```diff
@@ -20,8 +20,9 @@ public static T VerifyCompleted<T>(this ValueTask<T> task, string message = "Val
         {
             Contract.ThrowIfFalse(task.IsCompleted, message);
 
-            // Propagate any exceptions that may have been thrown.
-            return task.GetAwaiter().GetResult();
+            // Propagate any exceptions that may have been thrown. Relies on ValueTask.Result behaving like
+            // Task.GetAwaiter().GetResult() in the presence of exceptions.
+            return task.Result;
```

Comment:
i don't get it.  why was the old code wrong?

Reply:
➡️ The old code had the same behavior for the case where the `ValueTask<T>` is already completed, but @stephentoub has suggested on some other cases where I used `.GetAwaiter().GetResult()` that it was unnecessary and we can just use `.Result` for `ValueTask<T>`.

Reply:
➡️ I removed this change from the pull request.